src/components/SideBarComponents/SideBarItem.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+import { useState } from "react";
+import Arrow from "../../../public/svg/Arrow";
+import styles from "./SideBarItem.module.css";
+import Link from "next/link";
+
+function SideBarItem({
+  itemName,
+  elementsArray,
+  prefixPath,
+}: {
+  itemName: string;
+  elementsArray: { name: string; path: string }[];
+  prefixPath: string;
+}) {
+  const [itemOpened, setItemOpened] = useState<boolean>(false);
+  return (
+    <>
+      <div
+        className={styles.item}
+        onClick={() => {
+          setItemOpened(!itemOpened);
```

Comment:
do not override state this way better use setState((prev) => !prev)

Reply:
K